components/service/glean/src/main/java/mozilla/components/service/glean/TimespanMetricType.kt

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean
+
+import mozilla.components.service.glean.storages.TimespanResolution
+import mozilla.components.service.glean.storages.TimespansStorageEngine
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * This implements the developer facing API for recording timespans.
+ *
+ * Instances of this class type are automatically generated by the parsers at build time,
+ * allowing developers to record values that were previously registered in the metrics.yaml file.
+ *
+ * The timespans API exposes the [start], [stopAndSum] and [cancel] methods.
+ */
+data class TimespanMetricType(
+    override val disabled: Boolean,
+    override val category: String,
+    override val lifetime: Lifetime,
+    override val name: String,
+    override val sendInPings: List<String>,
+    val resolution: TimespanResolution
```

Comment:
This reminds me: we'll need to add an enum for the unit values over on the Python side.  That way the generated code will use enums and not strings.  (See the `lifetime` parameter for an example of how this works).